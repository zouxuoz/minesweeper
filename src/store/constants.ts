export enum GameStatus {
  PLAYING = 'PLAYING',
  WIN = 'WIN',
  LOSE = 'LOSE',
}

export const BOMB = 9;

export const START_GAME = 'START_GAME';
export const RESTART_GAME = 'RESTART_GAME';
export const RESET_GAME = 'RESET_GAME';
export const END_GAME = 'END_GAME';
export const REVEAL_CELL = 'REVEAL_CELL';
export const TOGGLE_FLAG = 'TOGGLE_FLAG';

export const SET_STATUS = 'SET_STATUS';
export const SET_CONFIG = 'SET_CONFIG';
export const SET_BOARD = 'SET_BOARD';
