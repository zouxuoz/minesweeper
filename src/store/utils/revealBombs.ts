import { BOMB } from '../constants';
import { GameState } from '../types';

export const revealBombs = (board: GameState['board']): GameState['board'] => {
  return board.map(row =>
    row.map(cell => (cell.value === BOMB ? { ...cell, revealed: true } : cell)),
  );
};
