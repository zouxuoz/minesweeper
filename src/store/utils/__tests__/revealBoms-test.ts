import { revealBombs } from '../revealBombs';
import { BOMB } from '../../constants';

it('Should reveal bombs.', () => {
  const board = [
    [
      { value: 1, revealed: false, flagged: false },
      { value: 3, revealed: false, flagged: false },
      { value: BOMB, revealed: false, flagged: false },
    ],
    [
      { value: 2, revealed: false, flagged: false },
      { value: BOMB, revealed: false, flagged: false },
      { value: BOMB, revealed: false, flagged: false },
    ],
    [
      { value: BOMB, revealed: false, flagged: false },
      { value: 3, revealed: false, flagged: false },
      { value: 2, revealed: false, flagged: false },
    ],
  ];

  expect(revealBombs(board)).toMatchSnapshot();
});
