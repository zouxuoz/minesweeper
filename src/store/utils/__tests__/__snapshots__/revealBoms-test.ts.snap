// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should reveal bombs. 1`] = `
Array [
  Array [
    Object {
      "flagged": false,
      "revealed": false,
      "value": 1,
    },
    Object {
      "flagged": false,
      "revealed": false,
      "value": 3,
    },
    Object {
      "flagged": false,
      "revealed": true,
      "value": 9,
    },
  ],
  Array [
    Object {
      "flagged": false,
      "revealed": false,
      "value": 2,
    },
    Object {
      "flagged": false,
      "revealed": true,
      "value": 9,
    },
    Object {
      "flagged": false,
      "revealed": true,
      "value": 9,
    },
  ],
  Array [
    Object {
      "flagged": false,
      "revealed": true,
      "value": 9,
    },
    Object {
      "flagged": false,
      "revealed": false,
      "value": 3,
    },
    Object {
      "flagged": false,
      "revealed": false,
      "value": 2,
    },
  ],
]
`;
